src/newmatrix.cpp

```diff
@@ -440,21 +458,13 @@ int system_test(rnum_t host, struct char_data *ch, int type, int software, int m
   target += modify_target_rbuf_raw(ch, rollbuf, sizeof(rollbuf), 8, FALSE) + DECKER->res_test + (DECKER->ras ? 0 : 4);
 
   for (struct obj_data *soft = DECKER->software; soft; soft = soft->next_content) {
-    if (GET_PROGRAM_TYPE(soft) == SOFT_SLEAZE)
-      detect = GET_PROGRAM_RATING(soft);
-    else if (!prog && GET_PROGRAM_TYPE(soft) == software) {
+    if (!prog && GET_PROGRAM_TYPE(soft) == software) {
       target -= GET_PROGRAM_RATING(soft);
       buf_mod(rollbuf, sizeof(rollbuf), "soft", -GET_PROGRAM_RATING(soft));
       prog = soft;
```

Comment:
Do you need to break here? Otherwise this will accept multiple copies of the same program and apply all of their bonuses.

Reply:
Done